src/Microsoft.AspNetCore.Server.Kestrel.Core/Adapter/Internal/RawStream.cs

```diff
@@ -115,17 +115,18 @@ public override Task FlushAsync(CancellationToken cancellationToken)
                 var readableBuffer = result.Buffer;
                 try
                 {
+                    if (result.IsCompleted)
+                    {
+                        return 0;
```

Comment:
If we want to make this similar to the request and response streams both read and write should throw.